### PR TITLE
Add 2ship-specific BUILDING.md and supportedHashes.json.

### PR DESCRIPTION
**Pull request by Malkierian**
null

### Build Artifacts
  - [2ship-mac.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1555146650.zip)
  - [2ship-linux.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1555152687.zip)
  - [2ship-windows.zip](https://nightly.link/HarbourMasters/2ship2harkinian/actions/artifacts/1555159396.zip)

### CODE REVIEW

File: docs/supportedHashes.json [hunk 1]
```diff
@@ -0,0 +1,10 @@
+[
+    {
+        "name": "N64 US",
+        "sha1": "328a1f1beba30ce5e178f031662019eb32c5f3b5"
```
Comment (inspectredc) [inline docs/supportedHashes.json]: i think this sha is wrong, mine is d6133ace5afaa0882cf214cf88daba39e266c078

File: docs/BUILDING.md [hunk 2]
```diff
@@ -0,0 +1,229 @@
...
+zypper in clang libstdc++-devel git cmake ninja SDL2-devel libpng16-devel libzip-devel libzip-tools nlohmann_json-devel tinyxml2-devel spdlog-devel
+```
+
+### Build
+
+_Note: If you're using Visual Studio Code, the [CMake Tools plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools) makes it very easy to just press run and debug._
+
+```bash
+# Clone the repo and enter the directory
+git clone https://github.com/HarbourMasters/Shipwright.git
```
Comment (sitton76) [inline docs/BUILDING.md:127]: This should be:

`git clone https://github.com/HarbourMasters/2ship2harkinian.git`

And the line under should be:

`cd 2ship2harkinian`

File: docs/BUILDING.md [hunk 3]
```diff
@@ -0,0 +1,229 @@
...
+1. Clone the 2 Ship 2 Harkinian repository
+
+_Note: Be sure to either clone with the ``--recursive`` flag or do ``git submodule init`` after cloning to pull in the libultraship submodule!_
+
+2. Place one or more [compatible](#compatible-roms) roms in the `OTRExporter` directory with namings of your choice
+
+_Note: Instructions assume using powershell_
+```powershell
+# Navigate to the Shipwright repo within powershell. ie: cd "C:\yourpath\Shipwright"
+cd Shipwright
```
Comment (inspectredc) [inline docs/BUILDING.md]:
```suggestion
# Navigate to the Shipwright repo within powershell. ie: cd "C:\yourpath\2ship2harkinian"
cd 2ship2harkinian
```

File: docs/BUILDING.md [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+Requires Xcode (or xcode-tools) && `sdl2, libpng, glew, ninja, cmake` (can be installed via homebrew, macports, etc)
+
+**Important: For maximum performance make sure you have ninja build tools installed!**
+
+_Note: If you're using Visual Studio Code, the [cpack plugin](https://marketplace.visualstudio.com/items?itemName=ms-vscode.cmake-tools) makes it very easy to just press run and debug._
+
+```bash
+# Clone the repo
+git clone https://github.com/HarbourMasters/Shipwright.git
+cd ShipWright
```
Comment (inspectredc) [inline docs/BUILDING.md]:
```suggestion
git clone https://github.com/HarbourMasters/2ship2harkinian.git
cd 2ship2harkinian
```

File: docs/BUILDING.md [hunk 5]
```diff
@@ -0,0 +1,229 @@
...
+& 'C:\Program Files\CMake\bin\cmake.exe' --build .\build\x64 --target Generate2ShipOtr
+```
+
+### Developing 2S2H
+With the cmake build system you have two options for working on the project:
+
+#### Visual Studio
+To develop using Visual Studio you only need to use cmake to generate the solution file:
+```powershell
+# Generates Ship.sln at `build/x64` for Visual Studio 2022
```
Comment (inspectredc) [inline docs/BUILDING.md]:
```suggestion
# Generates 2Ship.sln at `build/x64` for Visual Studio 2022
```

not sure on naming of this one

**Review by inspectredc — COMMENTED** [review]
Few more ship->2ship stuff

File: docs/supportedHashes.json (hunk 1 above)
Comment (Malkierian) [reply docs/supportedHashes.json]: I swear I pasted that...

**Review by garrettjoecox — APPROVED** [review]

File: docs/BUILDING.md [hunk 6]
```diff
@@ -0,0 +1,229 @@
...
+
+#### Regenerate Asset Headers
+```bash
+# If you need to regenerate the asset headers to check them into source
+cp <path to your ROM> OTRExporter
+cmake --build build-cmake --target ExtractAssetHeaders
+```
+
+## macOS
+Requires Xcode (or xcode-tools) && `sdl2, libpng, glew, ninja, cmake` (can be installed via homebrew, macports, etc)
```
Comment (inspectredc) [inline docs/BUILDING.md]: I think this needs these
```suggestion
Requires Xcode (or xcode-tools) && `sdl2, libpng, glew, ninja, cmake, nlohmann-json, libzip` (can be installed via homebrew, macports, etc)
```

**Comment by Dumbledork01** [comment]
Followed these steps and managed to get it to build. It may be beneficial to mention that downloading git separately is preferred rather than trying to pull the submodules in Visual Studio. Either I don't know how to do that in Visual Studio, or it is an absolute mess trying to do that lol.

**Pull request merged**